types/doubleclick-gpt/index.d.ts

```diff
@@ -168,7 +170,14 @@ declare namespace googletag {
         setTargeting(key: string, value: string | string[]): PubAdsService;
         setVideoContent(videoContentId: string, videoCmsId: string): void;
         updateCorrelator(): PubAdsService;
-        setPrivacySettings(settings: { restrictDataProcessing: boolean }): Slot;
+        setPrivacySettings(settings: PrivacySettingsConfig): Slot;
+    }
+
+    interface PrivacySettingsConfig {
+        childDirectedTreatment?: boolean;
+        limitedAds: boolean;
+        restrictDataProcessing: boolean;
```

Comment:
The docs appear to disagree on whether these `limitedAds`/`restrictDataProcessing` are required/optional. The examples here https://developers.google.com/publisher-tag/reference#example:_25 show that it's valid to call `setPrivacySettings` without setting `restrictDataProcessing` so that will need to be optional here. The example also shows `null` as a possible value that is functionally different from `undefined`, so please make these all optional AND nullable.

Reply:
Great input, thanks! I was hesitant to do it like this, but opted for the spec -- but I guess the spec is incomplete. Updated.

Reply:
looks good! thanks for contributing and being open to feedback 🙂